src/Workspaces/Core/Portable/Workspace/Host/DocumentService/IDocumentOperationService.cs

```diff
@@ -13,10 +13,5 @@ internal interface IDocumentOperationService : IDocumentService
         /// document version of <see cref="Workspace.CanApplyChange(ApplyChangesKind)"/>
         /// </summary>
         bool CanApplyChange { get; }
```

Comment:
dumb question.  why do we need this guy?  it even says you should be able to ask `"Workspace.CanApplyChange(ApplyChangesKind)"`

Reply:
➡️ Question moved to #34011

Reply:
I miss understood question since the issue only had that code snippet. it is there so that we want to provide a way for document provider to decide what kind of operation is supported per document provider since we want to make not workspace host does all the work to provide document, but provide a way for other such as razor for now, but like xaml or code generator to provide a document.

this is a way we will try to support changing generated code so that we can do code fix support for razor.

Reply:
we will not try to come up with final design, we will incrementally improve area here as we work with partner team. right now, every communication is done through IDocumentService and its specific services.